src/pepr/operator/controllers/keycloak/keycloak-client.ts

```diff
@@ -0,0 +1,397 @@
+/**
+ * Copyright 2025 Defense Unicorns
+ * SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+ */
+
+import { Client } from "./types";
+import { fetch, K8s, kind } from "pepr";
+import { Store } from "../../common";
+import { retryWithDelay } from "../utils";
+import { Component, setupLogger } from "../../../logger";
+import { FetchResponse } from "kubernetes-fluent-client/src/fetch";
+import { decodeJwt } from "jose";
```

Comment:
I haven't fully validated any of [these stackoverflow solutions](https://stackoverflow.com/questions/38552003/how-to-decode-jwt-token-in-javascript-without-using-a-library), but it would be good to remove this external dependency if we can and implement this JWT decoding on our own. Looking upstream at the [source](https://github.com/panva/jose/blob/08d1bd21eebf1fcd6a05c05e9e1dc469213b34bd/src/util/decode_jwt.ts#L31) for this I don't think it's overly complex to reimplement on our own (and the stackoverflow answers look similar).

There's in general no issue with us importing new libraries, but we should try to be critical and recreate/copy logic for simple things like this (<10 lines for some of those stackoverflow solves) to reduce our supply chain risks as much as possible.

Reply:
Done! Thanks for suggestion.